refactor(linter): Use DefaultRuleConfig for config options on three jest rules.

connorshea opened the pull request:
Just a simple refactor to get three more rules using the `DefaultRuleConfig` pattern. This also includes some documentation and diagnostics improvements.

camc314 commented:
### Merge activity

* **Dec 14, 1:52 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Dec 14, 1:52 PM UTC**: `camc314` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

connorshea committed:
refactor(linter): Use DefaultRuleConfig for config options on three jest rules. (#16814)

Just a simple refactor to get three more rules using the `DefaultRuleConfig` pattern. This also includes some documentation and diagnostics improvements.
